lib/cubit/webhook/lnurl_pay_service.dart

```diff
@@ -0,0 +1,172 @@
+  }
+
+  Future<Map<String, String>> _postWebhookRegistrationRequest({
+    required String pubKey,
+    required String webhookUrl,
+    required int currentTime,
+    required String signature,
+    String? lnAddressUsername,
+  }) async {
+    final String lnurlWebhookUrl = '$lnurlServiceURL/lnurlpay/$pubKey';
+    final Uri uri = Uri.parse(lnurlWebhookUrl);
+
+    final http.Response jsonResponse = await http.post(
+      uri,
+      body: jsonEncode(
+        AddWebhookRequest(
+          time: currentTime,
+          webhookUrl: webhookUrl,
+          username: lnAddressUsername,
+          signature: signature,
+        ).toJson(),
+      ),
+    );
+
+    if (jsonResponse.statusCode == 200) {
+      final Map<String, dynamic> data = jsonDecode(jsonResponse.body);
+      final String lnurl = data['lnurl'];
+      final String lnAddress = data.containsKey('lightning_address') ? data['lightning_address'] : '';
+      _logger.info('Registered LnUrl Webhook: $webhookUrl, lnurl = $lnurl, lnAddress = $lnAddress');
+      return <String, String>{
+        'lnurl': lnurl,
+        'lnAddress': lnAddress,
+      };
+    } else {
+      // TODO(erdemyerebasmaz): Handle username conflicts(only when user has created a new wallet)
+      // Add a random four-digit identifier, a discriminator, as a suffix if user's username is taken(~1/600 probability of conflict)
+      // Add a retry & randomizer logic until first registration succeeds
+      // TODO(erdemyerebasmaz): Handle custom username conflicts
+      throw Exception('Failed to register LnUrl Webhook: ${jsonResponse.body}');
+    }
+  }
+
+  Future<Map<String, String>> updateLnAddressUsername(WalletInfo walletInfo, String username) async {
+    final String? webhookUrl = await getLnUrlPayKey();
+    if (webhookUrl == null) {
+      throw Exception('Failed to retrieve registered webhook.');
+    }
+    return await registerLnurlpay(walletInfo, webhookUrl, username: username);
+  }
+
+  Future<void> _invalidateLnurlPay(String pubKey, String toInvalidate) async {
+    final String lnurlWebhookUrl = '$lnurlServiceURL/lnurlpay/$pubKey';
+    final int currentTime = DateTime.now().millisecondsSinceEpoch ~/ 1000;
+    final SignMessageRequest req = SignMessageRequest(message: '$currentTime-$toInvalidate');
+    final SignMessageResponse? signMessageRes = _breezSdkLiquid.instance?.signMessage(req: req);
+    if (signMessageRes == null) {
+      throw Exception('Missing signature');
+    }
+    final Uri uri = Uri.parse(lnurlWebhookUrl);
+    final http.Response response = await http.delete(
+      uri,
+      body: jsonEncode(
+        RemoveWebhookRequest(
+          time: currentTime,
+          webhookUrl: toInvalidate,
+          signature: signMessageRes.signature,
+        ).toJson(),
+      ),
+    );
+    _logger.info('invalidate lnurl pay response: ${response.statusCode}');
+    await resetLnUrlPayKey();
+    await resetLnAddressUsername();
```

Comment:
Do we need to reset the username? If `registerLnurlpay()` fails for some reason we can't get the username next time its called. Or is that the point?

Reply:
Yes, because assuming there was an error besides username conflict when registering, we'd like to avoid using that username the next time we're refreshing the webhook.

Reply:
@dangeross Actually, it's safe to remove this. We're not saving the username until registration succeeds.

It's removed on the new PR.